Tribler/community/tunnel/hidden_community.py

```diff
@@ -187,28 +224,21 @@ def monitor_downloads(self, dslist):
             old_state = self.download_states.get(info_hash, None)
             state_changed = new_state != old_state
 
-            time_elapsed = (time.time() - self.last_dht_lookup.get(info_hash, 0))
-            force_dht_lookup = time_elapsed >= self.settings.dht_lookup_interval
-
             # If the introducing circuit does not exist anymore or timed out: Build a new circuit
             if info_hash in self.infohash_ip_circuits:
-                for circuit_id in self.infohash_ip_circuits[info_hash]:
-                    if circuit_id not in self.my_intro_points:
-                        self.infohash_ip_circuits[info_hash].remove(circuit_id)
+                for (circuit_id, time_created) in self.infohash_ip_circuits[info_hash]:
+                    if circuit_id not in self.my_intro_points and time_created < time.time() - 30:
```

Comment:
Creating an introduction circuit should not cost more than 30 seconds I think? And if it does it is a slow circuit and will create a second one. I can increase the 30 seconds, but if one circuit generation fails, it takes even longer to create another one. What about 60 seconds?